ksonnet-gen/ksonnet/emit.go

```diff
@@ -46,6 +48,31 @@ type root struct {
 
 	ksonnetLibSHA *string
 	k8sSHA        *string
+	withNewSchema bool
+	traversed     []*kubespec.ParsedDefinitionName
+}
+
+func checkSchema(version string) bool {
```

Comment:
A couple notes about this.

1. I'd like to suggest that if we parse versions, we should use a "real" semver-parsing library. :)
2. It's actually not sufficient to "just" check semver matches. The release process for Kubernetes (particularly across major versions) implies significant changes to the API before the next version is released, but the actual version won't be incremented until release. In other words, it is possible to semantically match (say) ~v1.8.0 and actually be drastically incompatible with the existing rewrite rules, because we're about release v1.9.0.

So the question is what to do about these things?

For (1) we can nbd switch out the parsing code and vendor a dependency.

For (2) historically our (janky) solution has been to pin to _very_ specific versions of k8s, (_e.g._, v1.7.0 but not, say, v1.7.4). Since we expect ksonnet/ksonnet to pull _very_ specific versions of ksonnet-lib, it's probably not a bad idea to just keep doing this -- just error out if we aren't at _precisely_ the version SHA we expect.